### PR TITLE
Improve diff colors

### PR DESCRIPTION
**Pull request by thallada**
These new subtler diff colors which only set the background color so that syntax highlighting will still appear in the foreground colors.

Adds a set of "subtle" colors which can be used in either dark or light backgrounds.

I originally overrode these colors in my own config, see: https://github.com/ellisonleao/gruvbox.nvim/issues/105#issuecomment-1480547367

dark:

Screen Shot 2023-03-24 at 3 19 17 PM

light:

Screen Shot 2023-03-24 at 3 19 50 PM

### CODE REVIEW

File: lua/gruvbox/groups.lua [hunk 1]
```diff
@@ -148,10 +148,12 @@ M.setup = function()
...
-    DiffDelete = { fg = colors.red, bg = colors.bg0, reverse = config.inverse },
-    DiffAdd = { fg = colors.green, bg = colors.bg0, reverse = config.inverse },
-    DiffChange = { fg = colors.aqua, bg = colors.bg0, reverse = config.inverse },
-    DiffText = { fg = colors.yellow, bg = colors.bg0, reverse = config.inverse },
+    DiffDelete = { bg = colors.subtle_red },
+    DiffAdd = { bg = colors.subtle_green },
+    DiffChange = { bg = colors.subtle_brown },
+    DiffText = { bg = colors.subtle_yellow },
+    DiffviewDiffAddAsDelete = { bg = colors.subtle_red },
+    DiffviewDiffDelete = { bg = "none", fg = colors.bg2 },
```
Comment (ellisonleao) [inline lua/gruvbox/groups.lua:156]: why do we need `"none"` here?

Comment (thallada) [reply lua/gruvbox/groups.lua]: After doing some more testing I realized the Diffview highlights are not actually necessary so I removed them (they were only needed when I was overriding the old gruvbox theme).

**Comment by ellisonleao** [comment]
thanks @thallada

**Comment by mrvon** [comment]
@ellisonleao Hi，It would be better to add a option，it‘s weird in some colorscheme.

**Comment by eeeXun** [comment]
Agree, the color is not like gruvbox

**Comment by darkangel-ua** [comment]
I was forced to revert this change in my config - it's absolutely unreadable for me :(

**Comment by ellisonleao** [comment]
Looks like the diff is causing troubles hehe. We will revert this and think about a better color that can work for everyone. I will open an issue

**Comment by timsofteng** [comment]
I still doesn't see this cool improvement in diffview. Should I enable it with some option?

@thallada

**Comment by thallada** [comment]
Had to revert it since it had some issues. There's a discussion about bringing it back here: https://github.com/ellisonleao/gruvbox.nvim/discussions/239

**Comment by timsofteng** [comment]
Thanks. Got it. I'll track the issue. This is very helpful feature.

**Pull request merged**